Create `Guides/Finding Metadata` (#14455)
ppy/osu-wiki: wiki/Guides/Finding_Metadata/en.md

Metadata keywords can be used as filters in [song select](/wiki/Client/Interface#song-select) as well as on the website's [beatmap listing](https://osu.ppy.sh/beatmapsets) page via full-text search. The more precise and descriptive your metadata is, the easier it becomes for players to find your beatmap. Below is a quick workflow:

1. Start with what you already know: title fragments, artist name, lyrics, album art, or where you found the song.
2. Confirm the core fields (`Title`, `Artist`, and when relevant, `Source`) in primary sources. Note that if various sources show different versions, the one from the primary source should always be prioritised.
3. Use distribution and database sites to cross-check, find aliases, and identify practical keywords to use as tags.
4. Verify everything against [Ranking Criteria § Metadata](/wiki/Ranking_criteria/Metadata).

## Source reliability

Use sources in the order detailed in the table below whenever possible. The goal is to confirm final values from the most authoritative source available, then use lower-priority sources for cross-checking only.

| Priority | Source type | Examples |
| :-- | :-- | :-- |

> | 1 | Official/primary sources | Album liner notes and digital booklets, artist's official website and social medias like [Twitter](https://twitter.com), [Instagram](https://instagram.com), and [Facebook](https://facebook.com), official music videos and lyric videos on platforms like [YouTube](https://youtube.com) and [Niconico](https://nicovideo.jp) |

Reviewer: "social medias" is incorrect since "media" is already the plural of "medium"

```suggestion
| 1 | Official/primary sources | Album liner notes and digital booklets, artist's official website and social media presence like [Twitter](https://twitter.com), [Instagram](https://instagram.com), and [Facebook](https://facebook.com), official music videos and lyric videos on platforms like [YouTube](https://youtube.com) and [Niconico](https://nicovideo.jp) |
```

## Useful tips

### Use catalogue numbers for direct access

::: infobox
![Catalog number example](img/catalog-number.png "Catalog number of an album on VGMdb")
:::

Once you have identified a specific album or single, look for its [**catalog number**](https://en.wikipedia.org/wiki/Catalog_number_(music)). This is a unique identifier assigned by the record label (e.g., "SRCL-8641~2" for a Sony Music release). Catalog numbers can usually be found on metadata databases mentioned before.

Searching for this number directly (especially when combined with the label name) will often take you straight to the official product page on the label's or distributor's website.

### Navigating foreign-language sources

You can use the quick methods below when a source is not in your language:

- Try to use browser auto-translate if available.
- Use `Ctrl` + `F` / `Cmd` + `F` to find known names, titles, catalog numbers, or terms like "CD", "Music", "Tracklist", or "Discography".
- Check URLs and hover over links for recognisable English words before opening pages.

> 

Reviewer: Add a section about using [acoustic fingerprint identification](https://en.wikipedia.org/wiki/Acoustic_fingerprint) for audio, such as Shazam, MusicBrainz Picard or Google's search service (mobile only). Something like ``If you have no idea what the metadata is or where to start, try using <insert service here> or other similar services to get a clue as to what the metadata might be, then check that the song the service identified is the correct song by listening to it.``

- Musical information: genres, instruments (e.g., rock, piano, vocal chop)
- Additional credits: any artists not mentioned in the Artist field (e.g., featured vocalists, composers, remixers)
- Contextual information: aliases, series/franchise names, character names, featured artists, album name, remix information
- Beatmap-specific information: guest mapper usernames, contest names, hitsounders, storyboarders, etc.

For example, tags of beatmap [*流星群*](https://osu.ppy.sh/beatmapsets/1188437) can be broken down into the following groups:

::: infobox
![Example tag set](img/mapper-tags.png "Tag set for 流星群")
:::

- Event involved: `mappers' guild`, `mpg`
- Hitsounder: `ttmnzk`
- Featured artist tag: `featured artist`, `fa`
- Vocaloid: `vocaloid`, `hatsune miku`, `初音ミク`
- Artist alias: `cametak`, `kamelcamellia`
- Song title alias: `meteor shower`
- Genre and language: `dnb`, `drum and bass`, `electronic`, `japanese`
- Guest mappers: `celine`, `taku`, `voicecore`

### Touhou-specific conventions

> Touhou metadata has widely used community conventions. The checklist[^touhou-checklist] can be summarised as follows:

Author: Fixed in [a3ce50f](https://github.com/ppy/osu-wiki/pull/14455/commits/a3ce50f53e7dd9618a9eabc5ed50f855ab498b15)

Necessary ones:

| Tag group | Example | Notes |
| :-- | :-- | :-- |

> | media name | `touhou` |  |

Reviewer 2: ```suggestion
| Media name | `touhou` |  |
```

| media type | `video game` |  |

> | original song name | `Gensokyo Millenium 千年幻想郷　～ History of the Moon` |  |

Reviewer 2: ```suggestion
| Original song name | `Gensokyo Millenium 千年幻想郷　～ History of the Moon` |  |
```

| character name | `八意 永琳 eirin yagokoro` | add if the track is a boss theme |

> | vocalist for vocal arrangements | `ランコ ranko` |  |

Reviewer 2: ```suggestion
| Vocalist for vocal arrangements | `ランコ ranko` |  |
```

| album name | `TOHO COMPLETE BOX` |  |
| franchise alias | `東方project` |  |

Highly recommended additional ones:

| Tag group | Example(s) | Notes |
| :-- | :-- | :-- |
| creator/label credit | `ZUN Team Shanghai Alice` |  |
| event/convention name | `comiket c85`, `reitaisai` |  |
| characters involved | `marisa`, `alice` | characters in music or background |
| fan alias | `2hu` | which means "Touhou" |
| arranger name | `パインツリー` |  |

> | game number | `th8` | abbreviation for "Touhou 8" |

Reviewer 2: ```suggestion
| Game number | `th8` | Abbreviation for the edition in the series, here "Touhou 8" |
```

---
tags:
  - Metadata
  - tag
  - tags
---

# Finding metadata

> Metadata is data that defines and describes the characteristics of other data, and sometimes provides extra information.[^metadata] In the context of a [beatmap](/wiki/Beatmap), metadata includes elements such as the song's title, artist, source media, the beatmap's creator, guest mappers and more. These attributes help to describe the beatmap and its associated content. osu! provides a variety of metadata fields including `Title`, `Beatmap Creator` and `Tags`, which allow beatmaps to be effectively organised and discovered.

Author: I kind of don't want this line, because hosts can still add good metadata for beatmaps that aren't or are hardly going to enter a ranking procedure. 🥲

- **Vary your keyword combinations**: If your initial search doesn't yield the desired results, try different combinations of keywords or explore related search terms. For example, if you know the song is from a particular anime, game, or film, include that title in your search query to narrow down the results.
- **Follow the trail**: One source can often lead you to another, more authoritative source. A streaming platform page might link to the artist's official website or social media, which can provide more comprehensive and reliable metadata.
- **Build on what you find**: Use information uncovered in one search as keywords for the next. For example, if your initial search reveals the song's album name but not the artist, use that album name as a new keyword to find the artist. Each piece of discovered information becomes a tool to uncover the next.

## Common Sources

Once you've located potential sources, it's best to consult authoritative references to ensure accuracy. Below is a list of reliable options, categorized by their function:

> 1. Official & Primary Sources

Reviewer 2: How about this? 🤔 

<img width="1459" height="1321" alt="Image" src="https://github.com/user-attachments/assets/f022b20e-5fc9-4664-84b1-756c7e2ad523" />

- any known information such as the artist's name, or album name.

### Look for visual cues in URLs

Before clicking any link, hover over it and look at the URL displayed in your browser's status bar (usually at the bottom left). Even if the link text might be in a language you don't understand, the URL often contains recognisable English words like "music", "goods", "discography", "cd", or "product". This can guide you to the right section of the site without needing to read the text.

### Use catalogue numbers for direct access

Once you have identified a specific album or single, look for its **catalogue number**. This is a unique identifier assigned by the record label (e.g., "SRCL-8641~2" for a Sony Music release). Catalogue numbers can usually be found on databases like [VGMdb](https://vgmdb.net), [Discogs](https://www.discogs.com), or [MusicBrainz](https://musicbrainz.org).

Searching for this number directly (especially when combined with the label name) will often take you straight to the official product page on the label's or distributor's website, bypassing the need to navigate through menus in a foreign language.

## Source & Tags

> 

Reviewer 2: ```suggestion

See also: [Song setup](/wiki/Client/Beatmap_editor/Song_setup) and [Beatmap tags $ Mapper tags](/wiki/Beatmap/Beatmap_tags#mapper-tags)

```

---
tags:
  - Metadata
  - tag
  - tags
---

# Finding metadata

*See also: [Song setup § General](/wiki/Client/Beatmap_editor/Song_setup#general)*

Metadata is data that defines and describes the characteristics of other data, and sometimes provides extra information.[^metadata] In the context of a [beatmap](/wiki/Beatmap), this includes elements such as the song's title, artist, source media, the beatmap's creator, guest mappers and more. These attributes help to describe the beatmap and its associated content. Metadata can be added by the beatmap host during the [ranking process](/wiki/Beatmap_ranking_procedure). osu! provides a variety of metadata fields including `Title`, `Beatmap Creator` and `Tags`, all of which can be edited in the [Song setup](/wiki/Client/Beatmap_editor/Song_setup) window, allowing beatmaps to be effectively organised and discovered.

![Beatmap Listing](img/beatmap-listing.jpg "Screenshot of the beatmap listing, taken on 28th February 2026")

> Finding accurate metadata for your beatmap is, in essence, a process of [SEO](https://en.wikipedia.org/wiki/Search_engine_optimization) (Search Engine Optimisation). The more precise and descriptive your metadata is, the easier it becomes for players to find your beatmap. Metadata keywords can be used as filters in [song select](/wiki/Client/Interface#song-select) as well as on the website's [beatmap listing](https://osu.ppy.sh/beatmapsets) page via full-text search. Moreover, well-researched and detailed metadata increases the likelihood that your beatmap will attract modders, nominators, and ultimately progress through the ranking process more smoothly. Therefore, compiling accurate metadata is essential to ensure your beatmap reaches the right audience and receives the attention it deserves.

Reviewer 3: while the analogy might be true, it's rather niche to be a useful comparison

- **Symmetry**: The discovery process mirrors the search process. The keywords that led you to the song's metadata are likely the same terms players will use to find your beatmap. You can leverage this by incorporating those keywords into your metadata tags.
- **Relevance**: Focus on the most pertinent keywords players are likely to search for. For example, if your beatmap is a cover of a well-known song, consider including the original artist in the tags.

## Identifying music

First, you need to gather some basic information about the song. You may have discovered a track on platforms like [YouTube](https://www.youtube.com) or [SoundCloud](https://soundcloud.com) and obtained the MP3 from there. More often than not, you already know which song you want to map, or you've come across something interesting while browsing.

If you don't recognise the song, you can try identifying it using a music recognition app such as [Shazam](https://www.shazam.com) or [SoundHound](https://music.soundhound.com/soundhound). These tools can identify songs based on a short audio sample. If the song is recognised, the app will provide you with the title, artist and sometimes even album information.

> If the song remains unidentified, try searching using any lyrics you remember or by describing its characteristics (e.g., genre, mood, instrumentation) in a search engine. This approach can sometimes lead you to forums, music communities, or social media threads where others have successfully identified the track. Alternatively, you can upload a recording to platforms like [Reddit's r/NameThatSong](https://www.reddit.com/r/NameThatSong/) or [WatZatSong](https://www.watzatsong.com), where music enthusiasts can help identify it for you.

Reviewer 3: very odd to recommend on an osu! platform

If you've successfully identified the song or already know what you're looking for, the next step is to dig deeper and find accurate metadata through targeted searching.

> ## Collecting metadata

Author: I'm writing a whole rewrite of it now. Thanks...

- **Vary your keyword combinations**: If your initial search doesn't yield the desired results, try different combinations of keywords or explore related search terms. For example, if you know the song is from a particular anime, game, or film, include that title in your search query to narrow down the results.
- **Follow the trail**: One source can often lead you to another, more authoritative source. A streaming platform page might link to the artist's official website or social media, which can provide more comprehensive and reliable metadata.
- **Build on what you find**: Use information uncovered in one search as keywords for the next. For example, if your initial search reveals the song's album name but not the artist, use that album name as a new keyword to find the artist. Each piece of discovered information becomes a tool to uncover the next.

## Common Sources

Once you've located potential sources, it's best to consult authoritative references to ensure accuracy. Below is a list of reliable options, categorised by their function and sorted by priority:

> ### 1. Official & Primary Sources

Reviewer 3: why are these title-cased *and numbered* all of a sudden?

### 4. Music Metadata & Community Databases

These platforms are built specifically for collecting and organising release information. They are invaluable for cross-referencing and finding details for non-mainstream music.

- [**MusicBrainz**](https://musicbrainz.org): An open encyclopedia of music metadata. It's highly structured and aims to be a definitive authority.
- [**Discogs**](https://www.discogs.com): A comprehensive crowdsourced database of physical and digital releases. Excellent for finding various editions, label details, and catalog numbers.
- [**VGMdb**](https://vgmdb.net): A specialised database dedicated to video game and anime music. It is the go-to source for soundtracks and related works. Creating a free account on VGMdb also allows you to access user-submitted scans of CD booklets, obi strips, and disc faces, providing a level of detail that approaches the physical product itself.

### 5. Other Ranked Beatmaps

Checking the metadata of already ranked maps for the same song can be a useful shortcut. However, always verify the information against one of the official sources listed above to ensure its accuracy.

> ## Practical tips for navigating foreign language sites

Reviewer 3: these are useful tips. another one would be to auto-translate the page, either through google translate or related browser features, given that they exist

Searching for this number directly (especially when combined with the label name) will often take you straight to the official product page on the label's or distributor's website, bypassing the need to navigate through menus in a foreign language.

## Source & Tags

When filling out metadata, two fields play a particularly important role: `Source` and `Tags`.

The `Source` field should contain the original media the song was produced for, or the media it is most commonly associated with. This is often a game, anime, film, TV series, commercial or web series. If the song is a standalone single or not tied to any particular media, this field can be left blank. With all the information you've gathered, you should be able to fill out the `Source` field with confidence. If the song is associated with a specific media, make sure to use the official title of that media as it appears in your primary source.

The `Tags` field, on the other hand, allows you to include [a set of keywords](/wiki/Beatmap/Beatmap_tags#mapper-tags) that further describe the beatmap. This can include:

> - Musical information: genres, instruments, mood (e.g., rock, piano, emotional)

Author: not emotional, but I have seen ranked maps with tags like sentiment of the song.

> it'd be more useful to include an example of a beatmap's actual tags and explain where each of them come from

Sure.

- Contextual information: related characters or series names, album name, remix information
- Beatmap-specific information: difficulty characteristics, guest mapper usernames, contest names
- Additional credits: any artists not mentioned in the Artist field (e.g., featured vocalists, composers, remixers)

> and more. The `Tags` field is crucial for helping players discover your beatmap through search and filtering, so it's important to include relevant, accurate, consistent and complete tags based on the information you've gathered.

Reviewer 3: right. also redundant.

```suggestion
```
